addon/ng2/tasks/uninstall.js

```diff
@@ -0,0 +1,157 @@
+const chalk       = require('chalk');
+const path        = require('path');
+const fs          = require('fs');
+const fse         = require('fs-extra');
+const exec        = Promise.denodeify(require('child_process').exec);
+const packageJSON = path.resolve(process.cwd(), 'package.json');
+const nodeModules = path.resolve(process.cwd(), 'node_modules');
+
+module.exports = Task.extend({
+  command: '',
+  completionOKMessage: 'Successfully uninstalled.',
+  completionErrorMessage: 'Error uninstalling package.',
+
+  run: function(options) {
+    this.package = options.package;
+    this.typings = options.typings;
+
+    this.disableLogger();
+
+    this.ui.startProgress(chalk.green('Uninstalling package:', chalk.yellow(this.package)), chalk.green('.'));
+
+    return this.uninstallProcedure();
+  },
+
+  uninstallProcedure: function() {
+    const that = this;
+
+    return that.removeFromPackageJSON(that.package)
+    .then(() => {
+      return that.npmUninstall(that.package)
+      .then(() => {
+        return that.unlinkPackage(that.package)
+        .then(() => {
+          return that.removeFromSystemJSConfig(that.package)
+          .then(() => {
+            that.ui.stopProgress();
+            if (that.typings) {
+              that.ui.startProgress(chalk.green('Uninstalling typings:', chalk.yellow(that.typings)), chalk.green('.'));
+              return that.typingsUninstall(that.typings)
+              .then(() => {
+                that.ui.stopProgress();
+                that.announceOKCompletion();
+              });
+            }
+            else {
+              that.announceOKCompletion();
+            }
+          });
+        });
+      });
+    });
+  },
+
+  npmUninstall: function(pkg) {
+    let cmd = ['npm uninstall'];
+    const npmOptions = [
+      '--loglevel error',
+      '--color always',
+      '--save-dev'
+    ];
+    cmd.push(pkg);
+
+    return exec(cmd.concat(npmOptions).join(' '));
+  },
+
+  typingsUninstall: function(pkg) {
+    let cmd = ['typings uninstall'];
+    let options = [
+      '--ambient',
+      '--save'
+    ];
+    cmd.push(pkg);
+
+    return exec(cmd.concat(options).join(' '));
+  },
+
+  removeFromPackageJSON: function(pkg) {
+    const packageJSONContent = require(packageJSON);
+    const mainFile = require(path.resolve(nodeModules, pkg, 'package.json')).main;
+    let packages = require(packageJSON)['angular-cli'].packages || [];
+    
+    delete packages[pkg];
+
+    packageJSONContent['angular-cli'].packages = packages;
+    
+    fs.writeFileSync(packageJSON, JSON.stringify(packageJSONContent, null, '\t'), 'utf8');
+
+    return Promise.resolve();
+  },
+
+  unlinkPackage: function(pkg) {
+    const dir = path.resolve(process.cwd(), 'src', 'libs', pkg);
+    fse.removeSync(dir);
+
+    return Promise.resolve();
+  },
+
+  removeFromSystemJSConfig: function(pkg) {
+    const systemPath = path.resolve(process.cwd(), 'src', 'config', 'system.config.js');
```

Comment:
No, it works for plain `js` files... I added separate SystemJS config as you stated in the docs.